rfc/system/5606-json-idl.md

```diff
@@ -0,0 +1,343 @@
+}
+```
+
+reference: https://github.com/vmihailenco/msgpack 
+
+##### JavaScript
+```javascript
+import msgpack5 from 'msgpack5';
+
+// Create a MessagePack instance
+const msgpack = msgpack5();
+
+// Example data to encode
+const data = { a: 1 };
+
+// Encode the data
+const encodedData = msgpack.encode(data);
+
+// Print the encoded data
+console.log(encodedData); // <Buffer 81 a1 61 01>
+
+// Decode the data
+const decodedData = msgpack.decode(encodedData);
+
+// Print the decoded data
+console.log(decodedData); // { a: 1 }
+```
+reference: https://github.com/msgpack/msgpack-javascript 
+
+
+### FlyteIDL
+```proto
+message Json {
+    bytes value = 1;
+}
+
+message Scalar {
+    oneof value {
+        Primitive primitive = 1;
+        Blob blob = 2;
+        Binary binary = 3;
+        Schema schema = 4;
+        Void none_type = 5;
+        Error error = 6;
+        google.protobuf.Struct generic = 7;
+        StructuredDataset structured_dataset = 8;
+        Union union = 9;
+        Json json = 10; // New Type
+    }
+}
+```
+
+### FlytePropeller
+1. Attribute Access for dictionary, Datalcass, and Pydantic in workflow.
+Dict[type, type] is supported already, we have to support Datalcass and Pydantic now.
+```python
+from flytekit import task, workflow
+from dataclasses import dataclass
+
+@dataclass
+class DC:
+    a: int
+
+@task
+def t1() -> DC:
+    return DC(a=1)
+
+@task
+def t2(x: int):
+    print("x:", x)
+    return
+
+@workflow
+def wf():
+  o = t1()
+  t2(x=o.a)
+```
+2. Create a Literal Type for Scalar when doing type validation.
+```go
+func literalTypeForScalar(scalar *core.Scalar) *core.LiteralType {
+  ...
+  case *core.Scalar_Json:
+		literalType = &core.LiteralType\
+			{Type: &core.LiteralType_Simple{Simple: core.SimpleType_JSON}}
+  ...
+  return literalType 
+}
+```
+3. Support input and default input
+```go
+// Literal Input
+func ExtractFromLiteral(literal *core.Literal) (interface{}, error) {
+    switch literalValue := literal.Value.(type) {
+        case *core.Literal_Scalar:
+        ...
+            case *core.Scalar_Json:
+                return scalarValue.Json.Value, nil
+    }
+}
+// Default Input
+func MakeDefaultLiteralForType(typ *core.LiteralType) (*core.Literal, error) {
+    switch t := typ.GetType().(type) {
+	case *core.LiteralType_Simple:
+        ...
+        case core.SimpleType_JSON:
+                        return &core.Literal{
+                            Value: &core.Literal_Scalar{
+                                Scalar: &core.Scalar{
+                                    Value: &core.Scalar_Json{
+                                        Json: &core.Json{
+                                            Value: []byte("{}"),
+                                        },
+                                    },
+                                },
+                            },
+                        }, nil
+    }
+}
+```
+### FlyteKit
```

Comment:
Nit:

I feel this detail is not needed in the RFC, the addition of the Json message to flyteidl contains all the information.

Reply:
I think this is not needed yeah, we don't want to draw attention to model files in general.  But could you add an example showing how pyflyte run would work?